### ISSUE TITLE
Flake 8 set to exit zero?

### ISSUE THREAD

**Issue by shawaj** [NebraLtd/hm-diag#118]
Is flake 8 intentionally set to exit zero?

https://github.com/NebraLtd/hm-diag/blob/8b7df9a89851108eac6d177d3f4a5edb0cf6d08a/.github/workflows/python-tests.yml

As seemingly lots of failures...
https://github.com/nebraltd/hm-diag/runs/3357382650?check_suite_focus=true

**Comment by vpetersson** [comment]
Yeah i think we did this intentionally just to bring awareness to it.

**Comment by shawaj** [comment]
Surely defeats the object of the test if you just let it pass regardless?

**Comment by vpetersson** [comment]
Of course. It was just brought in so that we can enable it easily once we have fixed the issues.

**Comment by shawaj** [comment]
Fair enough!

**Comment by shawaj** [comment]
Fixed all the minor ones in #122

Only one left is:
```
Warning: ./hw_diag/tasks.py:22:1: C901 'perform_hw_diagnostics' is too complex (13)
```
